### PR TITLE
Code comments localized survey

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Per conversations with loc folks, find out non native speakers view English code comments.

[Internal review URL](https://review.docs.microsoft.com/en-us/dotnet/standard/io/pipelines?branch=pr-en-us-17129)

### CODE REVIEW

**Review by tdykstra — COMMENTED** [review]
Can you talk to loc and find out if they can put this blurb in localized versions only?

**Comment by Rick-Anderson** [comment]
Yes, we had that conversation. Currently they can't.

cc @JasonCard

**Comment by Thraka** [comment]
Such an odd PR 😄 Is there an OKR or something attached to this? Why these articles?

**Comment by Rick-Anderson** [comment]
I'd like everyone to sprinkle the include where appropriate. I worked on the buffers and pipes docs. Code comments are critical. I've been having discussions with loc about code comments. They can localized code comments, but they want customer feedback. Localized code comments could differentiate us from AWS docs (which currently have better CSAT in localized languages because they are translated by humans, not software).

Feel free to suggest other articles. I'd like to add it to high traffic articles that have important content in the code comments.

**Comment by tdykstra** [comment]
I'm reluctant to add a blurb that will just be a nuisance for the vast majority of people who use the docs.  Have you talked to Sara Lerner about other options for soliciting feedback from users of localized docs?

**Comment by JasonCard** [comment]
Since many international customers also read English version, we could also gather feedback on this from English live page. So I think we don’t need to hide it for English and we can rephrase it like “If you would like to see code comments translated to languages other than English, let us know in this GitHub discussion issue.”

**Comment by Rick-Anderson** [comment]
> I'm reluctant to add a blurb that will just be a nuisance for the vast majority of people who use the docs.

It's a trivial nuisance for English speakers but an ENORMOUS handicap for non-native. Let's try on on some samples that have critical info in the code comments.

**Comment by tdykstra** [comment]
OK, how long do you want to leave it in before removing it?

**Comment by Thraka** [comment]
If we want to gather feedback on international users and code comments in code blocks, why isn't the docs platform/publishing involved with adding a header or something to urls that are international and gathering feedback in a more controlled manner than through a generic github issue?

**Comment by Rick-Anderson** [comment]
They're considering it. We'd like to  get some data on this problem. For some topics, the code comments are critical.

> OK, how long do you want to leave it in before removing it?

A month or two.

**Comment by Rick-Anderson** [comment]
@Thraka  is an extremely minor inconvenience  for english readers. If it's more than that, they'll let us know in the issue. It's  most likely a significant issue for non-english folks. We need to get some feedback so we can use that to drive new tools, like the code comment translator.

**Comment by Rick-Anderson** [comment]
@Thraka  

> docs platform/publishing involved with adding a header or something to urls that are international and gathering feedback in a more controlled manner

That's likely to take a year or more. In the mean time, with feedback we could get a tool to translate code comments. Translated code comments could differentiate us from AWS.

This approach has the support of the loc folks.

Contrast the inconvenience of the survey to native speakers vs [Read a single message](https://docs.microsoft.com/en-us/dotnet/standard/io/pipelines#read-a-single-message) in the following format:

```
async ValueTask<Message> ReadSingleMessageAsync(PipeReader reader,
 CancellationToken cancellationToken = default)
{
    while (true)
    {
        ReadResult result = await reader.ReadAsync(cancellationToken);
        ReadOnlySequence<byte> buffer = result.Buffer;

        //如果沒有成功解析任何消息，請標記為消耗
        //為空，並作為整個緩衝區進行檢查。
        SequencePosition consumed = buffer.Start;
        SequencePosition examined = buffer.End;

        try
        {
            if (TryParseMessage(ref buffer, out Message message))
            {
                ///成功解析了一條消息，因此將開始標記為
                //解析的緩衝區已消耗。 TryParseMessage將緩衝區修剪為
                //解析消息後指向數據。
                consumed = buffer.Start;

                //已檢查標記為與此處消耗的相同，因此下一次通話
                //如果存在以下情況，則ReadSingleMessageAsync將處理下一條消息
                //一個
                examined = consumed;

                return message;
            }

            // 沒有更多數據要處理。
            if (result.IsCompleted)
            {
                if (buffer.Length > 0)
                {
                    // 消息不完整，沒有更多數據要處理。
                    throw new InvalidDataException("Incomplete message.");
                }

                break;
            }
        }
        finally
        {
            reader.AdvanceTo(consumed, examined);
        }
    }

    return null;
}
```

**Comment by Rick-Anderson** [comment]
@Thraka 
>  Is there an OKR or something attached to this? 

Yes,  inclusion is company wide D&I goal.

**Comment by Rick-Anderson** [comment]
> I'm reluctant to add a blurb that will just be a nuisance for the vast majority of people who use the docs. 

I don't think it's vast majority. @JasonCard  do you know what % of readers are english vs all others?

**Comment by JasonCard** [comment]
Yes, for .NET content, about 26% of all Page Views are for localized languages. So a little over a quarter.

**Comment by tdykstra** [comment]
@Thraka I'm OK with trying this if we specify a date one month or so out to assess the resulting feedback and revert the PR. What do you think?

@Rick-Anderson To make it easier to undo the code comments blurb when the time allotted for it expires, the PR shouldn't have unrelated changes in it.  I noticed an unrelated change to existing wording.

**Comment by Thraka** [comment]
I'm OK with it.

I don't, however, understand why we need to even solicit this feedback. It's common sense, it should just be translated. Is there a specific goal of "if X users vote yes, we'll make a tool?" That's like saying we're only going to translate articles into German when X users say they want it.

Let's set a date for removal, clean the PR to **ONLY** include adding the include items.

**Comment by JasonCard** [comment]
To provide more background, I've attached email from Paul Dempsey which explains the complexity of this feature request in [Customer Suggestion 149096: Feature request: Localize code comments](https://ceapex.visualstudio.com/Engineering/_workitems/edit/149096). Considering the complexity of this feature request, we are not sure if ROI would be worth the engineering time, cost, and risk. We do agree it would be nice to have localized comments, so we are requesting for investigation into how much dev effort would be needed. However, as we currently don’t have any data on this demand, it's considered low priority for now, so it may take some time to even start investigation on this. So if we can get some data on customer demand for this feature, we may be able to raise it's priority.

File: docs/standard/io/how-to-use-anonymous-pipes-for-local-interprocess-communication.md [hunk 1]
```diff
@@ -28,7 +28,9 @@ Anonymous pipes provide interprocess communication on a local computer. They off
   
  [!code-cpp[System.IO.Pipes.AnonymousPipeServerStream_Sample#01](../../../samples/snippets/cpp/VS_Snippets_CLR_System/system.IO.Pipes.AnonymousPipeServerStream_Sample/cpp/program.cpp#01)]
  [!code-csharp[System.IO.Pipes.AnonymousPipeServerStream_Sample#01](../../../samples/snippets/csharp/VS_Snippets_CLR_System/system.IO.Pipes.AnonymousPipeServerStream_Sample/cs/Program.cs#01)]
- [!code-vb[System.IO.Pipes.AnonymousPipeServerStream_Sample#01](../../../samples/snippets/visualbasic/VS_Snippets_CLR_System/system.IO.Pipes.AnonymousPipeServerStream_Sample/vb/program.vb#01)]  
+ [!code-vb[System.IO.Pipes.AnonymousPipeServerStream_Sample#01](../../../samples/snippets/visualbasic/VS_Snippets_CLR_System/system.IO.Pipes.AnonymousPipeServerStream_Sample/vb/program.vb#01)]
```
Comment (Thraka) [inline docs/standard/io/how-to-use-anonymous-pipes-for-local-interprocess-communication.md]: Lets revert this back to double spaced to clean up this PR

```suggestion
 [!code-vb[System.IO.Pipes.AnonymousPipeServerStream_Sample#01](../../../samples/snippets/visualbasic/VS_Snippets_CLR_System/system.IO.Pipes.AnonymousPipeServerStream_Sample/vb/program.vb#01)]  
```

File: docs/standard/io/how-to-write-text-to-a-file.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ The following classes and methods are typically used to write text to a file:
 - <xref:System.IO.Path> is for strings that have file or directory path information. It contains the <xref:System.IO.Path.Combine%2A> method and, in .NET Core 2.1 and later, the <xref:System.IO.Path.Join%2A> and <xref:System.IO.Path.TryJoin%2A> methods, which allow concatenation of strings to build a file or directory path.
 
 > [!NOTE]
-> The following examples show only the minimum amount of code needed. A real-world app usually provides more robust error checking and exception handling.  
+> The following examples show only the minimum amount of code needed. A real-world app usually provides more robust error checking and exception handling.
```
Comment (Thraka) [inline docs/standard/io/how-to-write-text-to-a-file.md]: Revert to clean up PR

```suggestion
> The following examples show only the minimum amount of code needed. A real-world app usually provides more robust error checking and exception handling.  
```

**Review by tdykstra — APPROVED** [review]

**Pull request merged**